src/main/kotlin/org/rust/lang/core/psi/ext/RsTraitType.kt

```diff
@@ -0,0 +1,10 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.psi.ext
+
+import org.rust.lang.core.psi.RsTraitType
+
+val RsTraitType.isImpl: Boolean get() = (stub?.isImpl) ?: impl != null
```

Comment:
Oh, it's my favorite part of kotlin operator precedence 😃 
I think you meant `stub?.isImpl ?: (impl != null)` because `(stub?.isImpl) ?: impl != null` is same as `((stub?.isImpl) ?: impl) != null`